### PR TITLE
WIP: Only give points that are available from the exercise

### PR DESCRIPTION
**Pull request by josalmi**
We don't want this to happen: http://u.tktl.fi/png/e2329b31ce2ee722bc2196b6e1832669

### CODE REVIEW

File: spec/lib/test_run_grader_spec.rb [hunk 1]
```diff
@@ -81,6 +81,27 @@ def successful_results
     expect(points).not_to include('1.2')
   end
 
+  it 'should only award points which are available from the exercise' do
+    otherExercise = FactoryGirl.create(:exercise, course_id: @submission.course_id)
+    FactoryGirl.create(:available_point, exercise_id: otherExercise.id, name: '1.3')
+
+    results = [
+      {
+        'className' => 'MyTest',
```
Comment (kasper) [inline spec/lib/test_run_grader_spec.rb]: Old hash-syntax.

**Comment by mpartel** [comment]
I take it you ran all the tests this time? :)

File: lib/test_run_grader.rb [hunk 2]
```diff
@@ -86,10 +86,11 @@ def self.award_points(submission, results, review_points)
     user = submission.user
     exercise = submission.exercise
     course = exercise.course
+    available_points = exercise.available_points.to_a.map(&:name)
```
Comment (jamox) [inline lib/test_run_grader.rb]: Why `.to_a`?

**Comment by jamox** [comment]
LGTM, except for the old hash syntax, can we change it? (or do we need to stick with it for the string keys)

**Pull request merged**